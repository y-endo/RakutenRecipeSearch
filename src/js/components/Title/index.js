import React from 'react';
import PropTypes from 'prop-types';

export default class Title extends React.Component {
  render() {
    return <h1 className="title">{this.props.title}</h1>;
  }
}

Title.propTypes = {
  title: PropTypes.string
};
