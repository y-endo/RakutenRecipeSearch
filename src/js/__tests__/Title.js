import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Title from '@/js/components/Title/index';

test('タイトルのテキストが正しく表示されるか', () => {
  const wrapper = shallow(<Title title="タイトル" />);

  // Titleのテキストが「タイトル」になっているか
  expect(wrapper.text()).toBe('タイトル');
  expect(toJson(wrapper)).toMatchSnapshot();

  // propsのtitleを「テスト」に変更
  wrapper.setProps({ title: 'テスト' });

  // Titleのテキストが「テスト」になっているか
  expect(wrapper.text()).toBe('テスト');
  expect(toJson(wrapper)).toMatchSnapshot();
});
