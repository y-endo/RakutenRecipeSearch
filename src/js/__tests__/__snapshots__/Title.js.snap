// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`タイトルのテキストが正しく表示されるか 1`] = `
<h1
  className="title"
>
  タイトル
</h1>
`;

exports[`タイトルのテキストが正しく表示されるか 2`] = `
<h1
  className="title"
>
  テスト
</h1>
`;
